lib/internal/repl.js

```diff
@@ -84,18 +90,26 @@ function setupHistory(repl, historyPath, ready) {
     if (err) {
       return ready(err);
     }
-
-    if (data) {
+    if (data && data[0] === '['
+        // Check the last two characters in case there is a trailing newline
+        && /\]/.test(data.substr(-2))) {
       try {
         repl.history = JSON.parse(data);
         if (!Array.isArray(repl.history)) {
           throw new Error('Expected array, got ' + typeof repl.history);
         }
         repl.history.slice(-repl.historySize);
```

Comment:
btw it feels wrong to slice it here. Why isn't it done before saving the file in `flushHistory ` ?